src/devices/src/bus.rs

```diff
@@ -197,21 +197,25 @@ mod tests {
     fn bus_insert() {
         let mut bus = Bus::new();
         let dummy = Arc::new(Mutex::new(DummyDevice));
-        assert!(bus.insert(dummy.clone(), 0x10, 0).is_err());
-        assert!(bus.insert(dummy.clone(), 0x10, 0x10).is_ok());
-
-        let result = bus.insert(dummy.clone(), 0x0f, 0x10);
-        assert!(result.is_err());
-        assert_eq!(format!("{:?}", result), "Err(Overlap)");
-
-        assert!(bus.insert(dummy.clone(), 0x10, 0x10).is_err());
-        assert!(bus.insert(dummy.clone(), 0x10, 0x15).is_err());
-        assert!(bus.insert(dummy.clone(), 0x12, 0x15).is_err());
-        assert!(bus.insert(dummy.clone(), 0x12, 0x01).is_err());
-        assert!(bus.insert(dummy.clone(), 0x0, 0x20).is_err());
-        assert!(bus.insert(dummy.clone(), 0x20, 0x05).is_ok());
-        assert!(bus.insert(dummy.clone(), 0x25, 0x05).is_ok());
-        assert!(bus.insert(dummy.clone(), 0x0, 0x10).is_ok());
+        for tc in &vec![
+            (0x10, 0, false, "Err(Overlap)"), // insert len should not be 0.
+            (0x10, 0x10, true, ""),
+            (0x0f, 0x10, false, "Err(Overlap)"), // this overlaps to address space of exist bus device based on 0x10
```

Comment:
nit: same-line comments make lines pretty long, might look better with comments on the line above the relevant code-line

Reply:
done.